### PR TITLE
Code Insights: Reuse series picker in compute insight creation UI page

### PR DESCRIPTION
**Pull request by vovakulikov**
Part of https://github.com/sourcegraph/sourcegraph/issues/37965
Fixes https://github.com/sourcegraph/sourcegraph/issues/38132 https://github.com/sourcegraph/sourcegraph/issues/37785

## Background 
Screenshot 2022-07-05 at 13 29 27

This PR adopts the 'FormSeries' component and reuses it on the compute-powered insight page.

## Test plan
- Make sure that all interactions with data series on the search-based creation UI page work correctly (edit data series, remove, add new one)
- Make sure that this logic works in the same way for the compute-powered insight creation UI

## App preview:

- [Web](https://sg-web-vk-re-use-series-picker-for.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-wboejamnyl.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/enterprise/insights/components/creation-ui/form-series/components/form-series-input/FormSeriesInput.tsx [hunk 1]
```diff
@@ -1,54 +1,42 @@
...
-
-    if (!isNotContext) {
-        return 'The `context:` filter is not supported; instead, run over all repositories and use the `context:` on the filter panel after creation'
-    }
-
-    if (!isNotRepo) {
-        return 'Do not include a `repo:` filter; add targeted repositories above, or filter repos on the filter panel after creation'
-    }
-}
+import { requiredNameField, validQuery } from './validators'
```
Comment (unclejustin) [inline client/web/src/enterprise/insights/components/creation-ui/form-series/components/form-series-input/FormSeriesInput.tsx:14]: 🙌 great clean-up!

File: client/web/src/enterprise/insights/components/creation-ui/form-series/components/form-series-input/FormSeriesInput.tsx [hunk 2]
```diff
@@ -1,54 +1,42 @@
...
-     */
+    /** Show all validation error of all fields within the form. */
     showValidationErrorsOnMount?: boolean
 
-    series: EditableDataSeries
-
-    /** Code Insight repositories field string value - repo1, repo2, ... */
+    /**
+     * Code Insight repositories field string value - repo1, repo2, ...
+     * This prop is used in order to generate a proper link for the query preview button.
```
Comment (unclejustin) [inline client/web/src/enterprise/insights/components/creation-ui/form-series/components/form-series-input/FormSeriesInput.tsx:27]: Thank you for adding the the use case

File: client/web/src/enterprise/insights/components/creation-ui/form-series/use-editable-series.ts [hunk 3]
```diff
@@ -2,61 +2,63 @@ import { useState } from 'react'
...
+    /**
+     * Call whenever the user cancel series editing by clicking cancel button
+     * in series form.
+     */
     cancelEdit: (seriesId: string) => void
+
+    /**
+     * Call whenever the user tries to delete series by clicking delete button
+     * in series preview card.
+     */
```
Comment (unclejustin) [inline client/web/src/enterprise/insights/components/creation-ui/form-series/use-editable-series.ts:52]: Thanks for all these clarifying comments 😄 ! Future me really appreciates it.

File: client/web/src/enterprise/insights/components/creation-ui/form-series/use-editable-series.ts [hunk 4]
```diff
@@ -163,11 +165,21 @@ export function useEditableSeries(props: UseEditableSeriesProps): UseEditableSer
...
+
+/** Helper replace element in array by index and return new array. */
+function replace<Element>(list: Element[], index: number, newElement: Element): Element[] {
+    return [...list.slice(0, index), newElement, ...list.slice(index + 1)]
+}
+
+/** Helper remove element from array by index. */
+function remove<Element>(list: Element[], index: number): Element[] {
+    return [...list.slice(0, index), ...list.slice(index + 1)]
+}
```
Comment (unclejustin) [inline client/web/src/enterprise/insights/components/creation-ui/form-series/use-editable-series.ts:185]: Was really surprised Lodash didn't have suitable functions for this already. Tsk tsk Lodash.

**Review by unclejustin — APPROVED** [review]

File: client/web/src/enterprise/insights/components/form/query-input/InsightQueryInput.module.scss [hunk 5]
```diff
@@ -5,6 +5,11 @@
 }
 
 .input-wrapper {
+    // Spread standard input paddings in order to fix visually problem
+    // with codemirror editor on the code insight creation UI pages.
+    // See https://github.com/sourcegraph/sourcegraph/issues/37785
+    padding-top: 0.5rem;
+    padding-bottom: 0.5rem;
```
[the remark this answers is not in the document]
Comment (vovakulikov) [reply client/web/src/enterprise/insights/components/form/query-input/InsightQueryInput.module.scss:12]: This just slightly spread Monaco input height but does not change this input dramatically.

[the remark this answers is not in the document]
Comment (vovakulikov) [reply client/web/src/enterprise/insights/components/form/query-input/InsightQueryInput.module.scss:12]: In fact in the screenshot in this PR description you can see UI with Monaco editor in data form series.

**Pull request merged**